tests/core/test_actions.py

```diff
@@ -762,4 +757,61 @@ def test_get_form_action_if_not_in_forms():
     )
 
     with pytest.raises(ActionNotFoundException):
-        assert not action.action_for_name(form_action_name, domain, None)
+        assert not action.action_for_name_or_text(form_action_name, domain, None)
+
+
+@pytest.mark.parametrize(
+    "end_to_end_utterance", ["Hi", f"{UTTER_PREFIX} is a dangerous start"]
+)
+def test_get_end_to_end_utterance_action(end_to_end_utterance: Text):
+    domain = Domain.from_yaml(
+        f"""
```

Comment:
this should use `textwrap.dedent`, the yaml isn't properly indented (top level keys with leading spacing - I guess it works, but we shouldn't bet on it)

Reply:
@wochinge could you please take this over